core/index/src/main/java/mil/nga/giat/geowave/core/index/CompoundIndexStrategy.java

```diff
@@ -0,0 +1,377 @@
+		final ByteBuffer buf = ByteBuffer.wrap(bytes);
+		buf.put(id1.getBytes());
+		buf.put(id2.getBytes());
+		buf.putInt(id1.getBytes().length);
+		return new ByteArrayId(
+				bytes);
+	}
+
+	/**
+	 * Get the ByteArrayId for each sub-strategy from the ByteArrayId for the
+	 * compound index strategy
+	 *
+	 * @param id
+	 *            the compound ByteArrayId
+	 * @return the ByteArrayId for each sub-strategy
+	 */
+	public ByteArrayId[] decomposeByteArrayId(
+			final ByteArrayId id ) {
+		final ByteBuffer buf = ByteBuffer.wrap(id.getBytes());
+		final int id1Length = buf.getInt(id.getBytes().length - 4);
+		final byte[] bytes1 = new byte[id1Length];
+		final byte[] bytes2 = new byte[id.getBytes().length - id1Length - 4];
+		buf.get(bytes1);
+		buf.get(bytes2);
+		return new ByteArrayId[] {
+			new ByteArrayId(
+					bytes1),
+			new ByteArrayId(
+					bytes2)
+		};
+	}
+
+	private List<ByteArrayId> composeByteArrayIds(
+			final List<ByteArrayId> ids1,
+			final List<ByteArrayId> ids2 ) {
+		final List<ByteArrayId> ids = new ArrayList<>(
+				ids1.size() * ids2.size());
+		for (final ByteArrayId id1 : ids1) {
+			for (final ByteArrayId id2 : ids2) {
+				ids.add(composeByteArrayId(
+						id1,
+						id2));
+			}
+		}
+		return ids;
+	}
+
+	private ByteArrayRange composeByteArrayRange(
+			final ByteArrayRange rangeOfStrategy1,
+			final ByteArrayRange rangeOfStrategy2 ) {
+		final ByteArrayId start = composeByteArrayId(
+				rangeOfStrategy1.getStart(),
+				rangeOfStrategy2.getStart());
+		final ByteArrayId end = composeByteArrayId(
+				rangeOfStrategy1.getEnd(),
+				rangeOfStrategy2.getEnd());
+		return new ByteArrayRange(
+				start,
+				end);
+	}
+
+	private List<ByteArrayRange> getByteArrayRanges(
+			final List<ByteArrayRange> ranges1,
+			final List<ByteArrayRange> ranges2 ) {
+		final List<ByteArrayRange> ranges = new ArrayList<>(
+				ranges1.size() + ranges2.size());
+		for (final ByteArrayRange range1 : ranges1) {
+			for (final ByteArrayRange range2 : ranges2) {
+				final ByteArrayRange range = composeByteArrayRange(
+						range1,
+						range2);
+				ranges.add(range);
+			}
+		}
+		return ranges;
+	}
+
+	@Override
+	public List<ByteArrayRange> getQueryRanges(
+			final MultiDimensionalNumericData indexedRange ) {
+		final MultiDimensionalNumericData[] ranges = getRangesForIndexedRange(indexedRange);
+		final List<ByteArrayRange> rangeForStrategy1 = subStrategy1.getQueryRanges(ranges[0]);
+		final List<ByteArrayRange> rangeForStrategy2 = subStrategy2.getQueryRanges(ranges[1]);
+		final List<ByteArrayRange> range = getByteArrayRanges(
+				rangeForStrategy1,
+				rangeForStrategy2);
+		return range;
+	}
+
+	@Override
+	public List<ByteArrayRange> getQueryRanges(
+			final MultiDimensionalNumericData indexedRange,
+			final int maxEstimatedRangeDecomposition ) {
+		final int maxEstRangeDecompositionPerStrategy = (int) Math.ceil(Math.sqrt(maxEstimatedRangeDecomposition));

```

Comment:
also, I'd argue if the argument is 'maxEstimatedRangeDecomposition' that you'd take the floor instead of the ceiling to try to keep the result less than max rather than more than max, but I could see arguments either way I suppose